src/devtools/client/debugger/src/reducers/breakpoints.ts

```diff
@@ -420,6 +434,54 @@ const customAnalysisResultComparator = (
   return result;
 };
 
+export const getAnalysisMappingForLocation = (state: UIState, location: Location | null) => {
+  if (!location) {
+    return undefined;
+  }
+  const locationKey = getLocationKey(location);
+  return state.breakpoints.analysisMappings.entities[locationKey];
+};
+
+export const getStatusFlagsForAnalysisEntry = (
+  analysisEntry: AnalysisRequest,
+  focusRegion: FocusRegion | null
+) => {
+  const { error, timeRange, status, points = [] } = analysisEntry;
+
+  const analysisErrored = [
+    AnalysisError.TooManyPointsToFind,
+    AnalysisError.TooManyPointsToRun,
+  ].includes(error!);
+
+  const analysisOverflowed =
+    status === AnalysisStatus.PointsRetrieved && points.length > MAX_POINTS_FOR_FULL_ANALYSIS;
+
+  const analysisLoaded = [AnalysisStatus.PointsRetrieved, AnalysisStatus.Completed].includes(
+    status
+  );
+
+  // TODO Drop this conversion when we convert FocusRegion to TimeStampedPointRange
+  const rangeAsFocusRegion: UnsafeFocusRegion = {
```

Comment:
btw - there is a function `rangeForFocusRegion` specifically made so you don't have to do this by hand 😄